fastlane/lib/fastlane/fastfile_parser.rb

```diff
@@ -0,0 +1,334 @@
+require 'parser'
+require 'parser/current'
+require 'pp'
+require "terminal-table"
+module Fastlane
+  class FastfileParser
+    attr_accessor :original_action
+
+    def to_s
+      "REMOVE_IT"
+    end
+
+    def lines
+      @lines ||= []
+    end
+
+    def lanes
+      @lanes ||= []
+    end
+
+    def actions
+      @actions ||= []
+    end
+
+    def counters
+      errors = lines.select { |key| key[:state] == :error }.length
+      deprecations = lines.select { |key| key[:state] == :deprecated }.length
+      infos = lines.select { |key| key[:state] == :infos }.length
+      { errors: errors, deprecations: deprecations, infos: infos, all: errors + infos + deprecations }
+    end
+
+    def data
+      { lanes: @lanes,  notices: @lines, actions: @actions }
+    end
+
+    def bad_options
+      [:use_legacy_build_api]
+    end
+
+    def redacted
+      rc = ""
+      rc = @content
+      self.class.secrets.each do |s|
+        rc.gsub!(s, "#" * s.length)
+      end
+      rc
+    end
+
+    # rubocop:disable PerceivedComplexity
```

Comment:
@KrauseFx  any idea on how to fix it? - i am not sure on how to split the whole thing up without reaching the `PerceivedComplexity`